### PR TITLE
Support solidjs libraries

### LINKED ISSUE #4871: @solid-js/router working in build and preview but not in dev
### What version of `astro` are you using?

1.3.0

### Are you using an SSR adapter? If so, which one?

none

### What package manager are you using?

pnpm

### What operating system are you using?

Windows

### Describe the Bug

When running the example simple routing with [@solidjs/router](https://github.com/solidjs/solid-router.git) example the workflow of `astro build` and `astro preview` is running fine. Running `astro dev` is failing with the error:
```
The requested module 'solid-js/web' does not provide an export named 'effect'
```
The error seems to originate from `solid-js/web` is imported with the `server` version which does not have an `effect` exported. I am struggling to find out why this is happening and how to work around it.

### Link to Minimal Reproducible Example

https://github.com/AirBorne04/minimal-astro-cloudflare-solid.git

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Fix #4871 

- Auto add solidjs deps to `ssr.noExternal` same as [vite-plugin-solid](https://github.com/solidjs/vite-plugin-solid/blob/03130c8a0a2ceaab9a07e16f1e1df832b996e1b8/src/index.ts#L251-L297)
- Support `__astro_tag_component__` in node_modules, since it imports `astro/server/index.js` in node_modules (doesn't work in pnpm). This PR pre-resolve `astro/server/index.js` to an absolute path.

## Testing

Added `@solidjs/router` to test, which previously would error the app entirely.

## Docs

Not sure if we should document the `noExternal` behaviour, it might be an implementation detail we want to hide.

### CODE REVIEW

File: packages/integrations/solid/src/dependencies.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		if (key === 'solid') return true;
+		if (typeof fields[key] === 'object' && containsSolidField(fields[key])) return true;
+	}
+	return false;
+}
+
+export function getSolidDeps(root: URL) {
+	const pkgPath = path.join(fileURLToPath(root), 'package.json');
+	if (!fs.existsSync(pkgPath)) {
+		console.log('No package.json found at project root');
```
Comment (matthewp) [inline packages/integrations/solid/src/dependencies.ts:23]: need need the eslint disable comment here

Comment (bluwy) [reply packages/integrations/solid/src/dependencies.ts:23]: Ah yeah I just copy-pasted this, will clean up the lints

File: packages/integrations/solid/src/dependencies.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		return [];
+	}
+	const require = createRequire(pkgPath);
+	const pkg = JSON.parse(fs.readFileSync(pkgPath, 'utf-8'));
+	const deps = [...Object.keys(pkg.dependencies || {}), ...Object.keys(pkg.devDependencies || {})];
```
[the remark this answers is not in the document]
Comment (bluwy) [reply packages/integrations/solid/src/dependencies.ts:28]: This part scans for the root project `package.json`, so I'm not sure if `peerDependencies` would make sense here 🤔

**Review by matthewp — APPROVED** [review]
lgtm, just needs to have the lockfile updated

**Pull request merged**